crates/llms/src/chat/mistral.rs

```diff
@@ -224,53 +223,103 @@ impl MistralLlama {
+        messages: RequestMessage::Completion {
+            text: prompt,
+            echo_prompt: false,
+            best_of: 1,
+        },
+        sampling_params: SamplingParams::default(),
+        response: tx,
+        return_logprobs: false,
+        is_streaming,
+        id: 0,
+        constraint: Constraint::None,
+        suffix: None,
+        adapters: None,
+    })
 }
 
 #[async_trait]
 impl Chat for MistralLlama {
+    async fn health(&mut self) -> Result<()> {
+        // If [`MistralLlama`] is instantiated successfully, it is healthy.
+        Ok(())
+    }
+
+    async fn stream<'a>(
+        &mut self,
+        prompt: String,
+    ) -> Result<Pin<Box<dyn Stream<Item = Result<Option<String>>> + Send>>> {
+        let (snd, mut rcv) = channel::<MistralResponse>(1000);
+        tracing::debug!("Sending request to pipeline");
+        self.pipeline
+            .get_sender()
+            .send(to_mistralrs_request(prompt, true, snd))
+            .await
+            .boxed()
+            .context(FailedToRunModelSnafu)?;
+        tracing::debug!("Request sent!");
+        Ok(Pin::from(Box::new(stream! {
+            while let Some(resp) = rcv.recv().await {
+                tracing::debug!("Received response from pipeline");
+                match resp {
+                    MistralResponse::Chunk(chunk) => {
+                        if let Some(choice) = chunk.choices.first() {
+                            yield Ok(Some(choice.delta.content.clone()));
+                            if choice.finish_reason.is_some() {
+                                break;
+                            }
+                        } else {
+                            yield Ok(None);
+                            break;
+                        }
+                    },
+                    MistralResponse::ModelError(err_msg, _) | MistralResponse::CompletionModelError(err_msg, _) => {
+                        yield Err(ChatError::FailedToRunModel {
+                            source: err_msg.into(),
+                        })
+                    },
+                    MistralResponse::InternalError(err_msg) | MistralResponse::ValidationError(err_msg) => {
+                        yield Err(ChatError::FailedToRunModel {
+                            source: err_msg,
+                        })
+                    },
+                    MistralResponse::CompletionDone(cr) => {
+                        yield Ok(Some(cr.choices[0].text.clone()));
+                        break;
+                    },
+                    MistralResponse::Done(_) => {
+                        unreachable!()
+                    },
+                }
+            }
+        })))
+    }
+
     async fn run(&mut self, prompt: String) -> Result<Option<String>> {
-        let r = self.to_request(prompt);
+        let (snd, mut rcv) = channel::<MistralResponse>(10_000);
+        tracing::debug!("Sending request to pipeline");
```

Comment:
nit: I would keep this at a `trace` level, since its only purpose is to help us trace the flow and doesn't contain any extra information from variables that would otherwise be useful for debugging.